test/e2e/suite/flake_attempt.go

```diff
@@ -0,0 +1,101 @@
+package suite
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+)
+
+type FlakeAttempt struct {
```

Comment:
nit wdty about
```suggestion
type FlakyCode struct {
```

so it reads something like `flakyCode.It(ctx, "does something", maxRetries`